src/App.tsx

```diff
@@ -14,11 +18,125 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  buttonVisible: boolean;
+  isReversed: boolean;
+  sortBy: string;
+};
+
+type Props = {};
+
+enum Sortby {
+  Name = 'name',
+  Length = 'length',
+}
+
+class App extends React.Component<Props, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    buttonVisible: true,
+    isReversed: false,
+    sortBy: '',
+  };
+
+  start = () => {
+    this.setState((state) => ({
+      buttonVisible: !state.buttonVisible,
+    }));
+  };
+
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sort = () => {
```

Comment:
better to use more informative names, also use suffix ('Handler') in naming to show that is function proceed some event